File: src/models/admin_auth_token_create_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminAuthTokenCreateIn {
    pub name: String,

    pub role: String,

    /// Milliseconds from now until the token expires.
    #[serde(
        rename = "expiry_ms",
        skip_serializing_if = "Option::is_none",
        with = "crate::duration_ms_serde::optional"
    )]
    pub expiry: Option<std::time::Duration>,

    /// Whether the token is enabled. Defaults to `true`.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub enabled: Option<bool>,
}

impl AdminAuthTokenCreateIn {
    pub fn new(name: String, role: String) -> Self {
        Self {
            name,
            role,
            expiry: None,
            enabled: None,
        }
    }

    pub fn with_expiry(mut self, value: impl Into<Option<std::time::Duration>>) -> Self {
        self.expiry = value.into();
        self
    }

    pub fn with_enabled(mut self, value: impl Into<Option<bool>>) -> Self {
        self.enabled = value.into();
        self
    }
}

File: coyote/models/admin_auth_token_create_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class AdminAuthTokenCreateIn(BaseModel):
    name: str

    role: str

    expiry_ms: t.Optional[int] = Field(default=None, alias="expiry_ms")
    """Milliseconds from now until the token expires."""

    enabled: t.Optional[bool] = None
    """Whether the token is enabled. Defaults to `true`."""
